daisy/step_sub_workflow.go

```diff
@@ -29,7 +29,7 @@ type SubWorkflow struct {
 func (s *SubWorkflow) populate(ctx context.Context, st *Step) DError {
 	if s.Path != "" {
 		var err error
-		if s.Workflow, err = st.w.NewSubWorkflowFromFile(s.Path); err != nil {
+		if s.Workflow, err = st.w.NewSubWorkflowFromFile(s.Path, s.Vars); err != nil {
```

Comment:
Is the loop over vars and adding them to the subworkflow below in this file needed, since it's already done by NewSubWorkflowFromFile?

Reply:
FromFile is not the only way. The sub/included workflow can actually be directly defined in json inline. Or, when we construct workflow in memory, similarly we can directly set a sub/included workflow.